com.unity.shadergraph/Editor/Data/Nodes/Math/Advanced/RefractNode.cs

```diff
@@ -0,0 +1,80 @@
+using System.Reflection;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+
+namespace UnityEditor.ShaderGraph
+{
+    enum RefractMode
+    {
+        CriticalAngle,
+        Safe
+    };
+
+    [Title("Math", "Advanced", "Refract")]
```

Comment:
I recommend putting this in the Math/Vector category instead of Math/Advanced.  It's very similar to the Reflection node - which is also in the Vector subcategory.